### PR TITLE
Update serving in mnist example; use 0.4 and add testing.

### PR DESCRIPTION
**Pull request by jlewi**
* Add the TFServing component
* Create TFServing components.

* The model.py code doesn't appear to be exporting a model in saved model
  format; it was a missing a call to export.

  * I'm not sure how this ever worked.

* It also looks like there is a bug in the code in that its using the cnn input fn even if the model is the linear one. I'm going to leave that as is for now.

* Create a namespace for each test run; delete the namespace on teardown
* We need to copy the GCP service account key to the new namespace.
* Add a shell script to do that.

Related to #460

---
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/examples/469)

### CODE REVIEW

**Comment by jlewi** [comment]
/assign @zhenghuiwang 
This is ready for review.

File: mnist/ks_app/app.yaml [hunk 1]
```diff
@@ -2,14 +2,28 @@ apiVersion: 0.3.0
 environments:
   jlewi:
     destination:
-      namespace: kubeflow
+      namespace: jlewi
```
Comment (zhenghuiwang) [inline mnist/ks_app/app.yaml]: Could we use a more meaningful namespace & path, such as `kf-mnist'?

File: mnist/ks_app/vendor/kubeflow/tf-serving@fed535eaa276220e4edf59530c0629f4375a40a9/README.md [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+* [Using Prototypes](#using-prototypes)
+  * [io.ksonnet.pkg.tf-serving](#io.ksonnet.pkg.tf-serving)
+
+## Quickstart
+
+*The following commands use the `io.ksonnet.pkg.tf-serving` prototype to generate Kubernetes YAML for tf-serving, and then deploys it to your Kubernetes cluster.*
+
+First, create a cluster and install the ksonnet CLI (see root-level [README.md](rootReadme)).
+
+If you haven't yet created a [ksonnet application](linkToSomewhere), do so using `ks init <app-name>`.
```
Comment (zhenghuiwang) [inline mnist/ks_app/vendor/kubeflow/tf-serving@fed535eaa276220e4edf59530c0629f4375a40a9/README.md]: Add the link?

File: test/copy_secret.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+set -e
+SOURCE=$1
+DEST=$2
+NAME=$3
+
+usage() {
+	echo copy_secret "<source namepspace> <dest namespace> <secret name>"
+}
+
+if [ -z ${SOURCE} ]; then
```
Comment (zhenghuiwang) [inline test/copy_secret.sh]: how about using `||` to combine these 3 checking statements.

**Pull request merged**